cmd/observer/node_builder/observer_builder.go

```diff
@@ -839,16 +800,39 @@ func (builder *ObserverServiceBuilder) enqueuePublicNetworkInit() {
 				return nil, fmt.Errorf("could not register networking receive cache metric: %w", err)
 			}
 
-			msgValidators := publicNetworkMsgValidators(node.Logger, node.IdentityProvider, node.NodeID)
-
-			builder.initMiddleware(node.NodeID, publicLibp2pNode, msgValidators...)
-
-			// topology is nil since it is automatically managed by libp2p
-			net, err := builder.initNetwork(builder.Me, builder.Metrics.Network, builder.Middleware, nil, receiveCache)
+			net, err := p2pnet.NewNetwork(&p2pnet.NetworkConfig{
```

Comment:
```suggestion
			net, err := p2pnet.NewEngineNetwork(&p2pnet.NetworkConfig{
```

Reply:
The network serves as a comprehensive layer that integrates three interfaces within Flow. Renaming it as `NewNetworkEngine` would misrepresent its function, implying it's merely an engine component rather than an all-encompassing networking layer.